### DISCUSSION TITLE
Error al hacer mvn verify: URF_UNREAD_FIELD

### DISCUSSION THREAD

**Discussion by antobevi** [dds-jv/foro#196] [Consultas de herramientas] [answered]
Buenas noches, disculpen las molestias. Cuando hago mvn verify me tira lo siguiente y no se como solucionarlo:

![1](https://user-images.githubusercontent.com/48884370/116018287-1c59b880-a618-11eb-94f2-d8414cf34972.png)
![2](https://user-images.githubusercontent.com/48884370/116018297-2085d600-a618-11eb-931e-d45ce7cb27eb.png)
![3](https://user-images.githubusercontent.com/48884370/116018307-24195d00-a618-11eb-8e06-a70235c68e1f.png)

**Comment by nanderson87** [comment, marked as answer]
Buenas, 

Este error guarda relación con que el atributo `fecha` de la clase `Venta` nunca esta siendo usado (o leído). Esto es así? Es decir le estas asignado un valor dentro del constructor, pero después lo estas usando en algún método?


Por otra parte fijate que dentro de la clase `Venta` los atributos (o fields)  `items`, `metodo` y `fecha` no tienen ningún modificar de acceso por lo que por defecto son públicos. Esto quiere decir que pueden ser modificados/leídos desde afuera. Entiendo que son parte del estado interno de la clase por lo que te recomiendo ponerlos como privados (private) 

Ej.:
`private  MetodoPago metodo;`

**Reply by antobevi** [reply]
Hola!
El atributo fecha solo lo estoy usando en los tests cuando creo una instancia de venta y le asigno algún valor.

**Reply by nanderson87** [reply]
Claro, el code smell detectado entonces sigue teniendo sentido. Porque como te comente antes lo que esta verificando es que tenes al atributo `fecha` (el mismo se le puede estar asignado o no un valor) pero no se esta usando en ningún momento dentro de la clase `Venta`. Es decir no hay ningún método que este haciendo algo con ese atributo dentro de la clase, ya sea un método que expongas o un método privado.

Lo raro es que si lo deberías estar usando para resolver este requerimiento del enunciado:

> Ah, un requerimiento más: Macowins registra las ventas de estas prendas y necesita saber las ganancias de un determinado día. 

Este requerimiento lo pudiste hacer? Porque para solucionar el mismo, de alguna manera si vas a terminar usando la fecha de la venta para calcular la ganancia de determinado día.

Por último, te recomiendo que la fecha no la modeles como un `String` sino que uses el tipo `LocalDate` de java. Ya que presenta una mejor abstracción y  nos va a permitir hacer operaciones mas interesantes relacionadas a una fecha. Como por ejemplo saber si una fecha es igual a otra. Obtener el día, el mes, el año. Sumarle/sacarle unidades de tiempos, verificar si esa fecha esta dentro de un rango, etc.

**Reply by antobevi** [reply]
Ahhh claro, yo lo resolví pero no tomé en cuenta la fecha, sume los importes de todas las ventas que tenía en la lista y ya está, lo había entendido mal 🤦🏻 
Muchas gracias profe! Ahora modifico eso teniendo en cuenta las cosas que me marco :)

**Reply by nanderson87** [reply]
Genial, teniendo en cuenta eso seguramente vas a usar la fecha en algun punto y no debería marcarte más el code smell de que no estas usando el atributo.

**Reply by antobevi** [reply]
Si, ahora que lo arreglé ya no me lo marca mas, gracias!
